pkg/kube/secretcontroller/secretcontroller.go

```diff
@@ -242,11 +245,13 @@ func (c *Controller) addMemberCluster(secretName string, s *corev1.Secret) {
 			log.Infof("Cluster %s in the secret %s in namespace %s already exists",
 				clusterID, c.cs.remoteClusters[clusterID].secretName, s.ObjectMeta.Namespace)
 		}
+		c.Unlock()
```

Comment:
This doesn't look right. You have cases above that `continue`, which would miss this unlock call. Maybe you should move all of this to a function and use `defer c.Unlock`?

Reply:
Yeah - I missed the continue in my haste - thanks for catching.   For now I am just checking if this has any influence on the racetest failures introduced in the pilot multicluster code.  I don't think it does but don't have a good way to run the reacetest locally.   I am likely to abandon this PR but changed per your suggestion anyway.